node/src/network.rs

```diff
@@ -124,60 +163,108 @@ impl MeshNetworkClient {
     fn sender_for(
         &self,
         task_id: MpcTaskId,
-        participants: Vec<ParticipantId>,
+        start: Option<&MpcStartMessage>,
+        originator: ParticipantId,
     ) -> SenderOrNewChannel {
-        if self.deleted_channels.lock().unwrap().contains(&task_id) {
-            return SenderOrNewChannel::RemovedChannel;
-        }
-        let mut senders_for_tasks = self.senders_for_tasks.lock().unwrap();
-        match senders_for_tasks.entry(task_id) {
-            Entry::Occupied(entry) => SenderOrNewChannel::Existing(entry.get().clone()),
+        let drop_fn = {
+            let channels = self.channels.clone();
+            move || {
+                channels.lock().unwrap().senders.remove(&task_id);
+            }
+        };
+        let mut channels = self.channels.lock().unwrap();
+        let sender = match channels.senders.entry(task_id) {
+            Entry::Occupied(entry) => entry.get().clone(),
             Entry::Vacant(entry) => {
-                let (sender, receiver) = mpsc::channel(CHANNEL_SIZE);
+                let (sender, receiver) = mpsc::unbounded_channel();
                 entry.insert(sender.clone());
-                drop(senders_for_tasks); // release lock
-
-                let senders_for_tasks = self.senders_for_tasks.clone();
-                let deleted_channels = self.deleted_channels.clone();
-                let drop_fn = move || {
-                    deleted_channels.lock().unwrap().put(task_id, ());
-                    senders_for_tasks.lock().unwrap().remove(&task_id);
-                };
-
-                SenderOrNewChannel::NewChannel {
-                    sender,
-                    channel: NetworkTaskChannel {
+                let incomplete_channel = IncompleteNetworkTaskChannel { receiver };
+                if let Some((k, _)) = channels
+                    .channels_waiting_for_start
+                    .push(task_id, incomplete_channel)
+                {
+                    if k != task_id {
+                        // Keep channels_waiting_for_start a subset of senders.
+                        channels.senders.remove(&k);
```

Comment:
Could you add some comment on what this logic means? It is unclear to me what `k != task_id` is supposed to mean

Reply:
Added plenty of extra comments; indeed this was not properly documented.